File: platform-kotlin/src/main/kotlin/sgdea/platform/storage/ObjectStorage.kt
package sgdea.platform.storage

// P-03: toda implementación self-hosted debe operar sin conectividad saliente (P-10).
interface ObjectStorage {
    fun put(key: String, content: ByteArray)
    fun get(key: String): ByteArray
    fun exists(key: String): Boolean
    fun delete(key: String)
}

// Backend gestionado (SaaS): proveedor concreto sin decidir todavía.
class ManagedObjectStorage : ObjectStorage {
    override fun put(key: String, content: ByteArray) = TODO()
    override fun get(key: String): ByteArray = TODO()
    override fun exists(key: String): Boolean = TODO()
    override fun delete(key: String) = TODO()
}

// Backend autoalojado (on-premise, P-10): debe funcionar sin conectividad saliente.
class SelfHostedObjectStorage : ObjectStorage {
    override fun put(key: String, content: ByteArray) = TODO()
    override fun get(key: String): ByteArray = TODO()
    override fun exists(key: String): Boolean = TODO()
    override fun delete(key: String) = TODO()
}

File: platform-python/src/platform_python/object_storage.py
from abc import ABC, abstractmethod


# P-03: toda implementación self-hosted debe operar sin conectividad saliente (P-10).
class ObjectStorage(ABC):
    @abstractmethod
    def put(self, key: str, content: bytes) -> None: ...

    @abstractmethod
    def get(self, key: str) -> bytes: ...

    @abstractmethod
    def exists(self, key: str) -> bool: ...

    @abstractmethod
    def delete(self, key: str) -> None: ...


# Backend gestionado (SaaS): proveedor concreto sin decidir todavía.
class ManagedObjectStorage(ObjectStorage):
    def put(self, key: str, content: bytes) -> None:
        raise NotImplementedError

    def get(self, key: str) -> bytes:
        raise NotImplementedError

    def exists(self, key: str) -> bool:
        raise NotImplementedError

    def delete(self, key: str) -> None:
        raise NotImplementedError


# Backend autoalojado (on-premise, P-10): debe funcionar sin conectividad saliente.
class SelfHostedObjectStorage(ObjectStorage):
    def put(self, key: str, content: bytes) -> None:
        raise NotImplementedError

    def get(self, key: str) -> bytes:
        raise NotImplementedError

    def exists(self, key: str) -> bool:
        raise NotImplementedError

    def delete(self, key: str) -> None:
        raise NotImplementedError
